### ISSUE TITLE
question about coordinating two components

### ISSUE THREAD

**Issue by jvsteiner** [yo35/kokopu-react#10]
In my use case, I would like to have an interactive Chessboard in one part of the component tree, and a MoveText in another part.  I would like them to share the same Game state: ie. the user might interact with either of them, and I want the other one to be synchronized.  Is this something that this library is designed for/capable of?  If so, would it be possible to offer a few hints about how to do it?

**Comment by jvsteiner** [comment]
I ended up using redux

**Comment by yo35** [comment]
Hi,
Sorry for the late reply.

If you just need to synchronize a `NavigationBoard` and `Movetext`, it should be that complex. A minimal example could be like:

```
function NavBoardWithMoveText({ game }) {

  const [nodeId, setNodeId] = useState('start');
  
  return (
    <>
        <NavigationBoard game={game} nodeId={nodeId} onNodeIdChanged={setNodeId} />
        <Movetext game={game} selection={nodeId} interactionMode="selectMove" onMoveSelected={n => if (n !== undefined) { setNodeId(n); } } />
    </>
  );

}
```
Of course, in a real-life code, you may want to lift the `nodeId` higher in the component tree, possibly persisting it with redux or any other persistency mechanism, but the key points regarding component synchronization are here.

**Comment by jvsteiner** [comment]
Yes, you are right - I have about three consumers of the state stretched from one end of the tree to the other, so I went with redux, since I had been meaning to learn how to use it anyway.

**Comment by yo35** [comment]
I've added this example to the documentation, as it may correspond to a frequent use-case:

https://kokopu-react.yo35.org/docs/current/#/Live%20demo/Synchronize%20NavigationBoard%20and%20Movetext%20components
